src/shared/varlink.h

```diff
@@ -158,6 +159,9 @@ unsigned varlink_server_current_connections(VarlinkServer *s);
 
 int varlink_server_set_description(VarlinkServer *s, const char *description);
 
+int varlink_server_serialize(VarlinkServer *s, FILE *f, FDSet *fds);
+int varlink_server_deserialize_one(VarlinkServer *s, const char *value, FDSet *fds);
```

Comment:
In the interest of eventually making the varlink stuff public, I think we'd want this in a separate file so we can keep it private to systemd.

The other idea I had was adding varlink_server_get_socket_first() and varlink_server_get_socket_next() so that we can iterate over the sockets and then implement the serialization logic in pid1 on top of that.